networkx/utils/backends.py

```diff
@@ -441,15 +455,6 @@ def __call__(self, /, *args, backend=None, **kwargs):
         #     if (val := args[pos] if pos < len(args) else kwargs.get(gname)) is not None
         # }
 
-        if self._is_testing and self._automatic_backends and backend_name is None:
-            # Special path if we are running networkx tests with a backend.
-            return self._convert_and_call_for_tests(
-                self._automatic_backends[0],
-                args,
-                kwargs,
-                fallback_to_nx=self._fallback_to_nx,
-            )
-
```

Comment:
Note that I moved this code block to further below so that iterators of input graphs are turned into lists. This allows the "fallback to networkx" functionality to work even for iterator of graphs inputs (otherwise, the iterators would be consumed by the first call).